core/src/test/java/org/bitcoinj/base/BitcoinNetworkTest.java

```diff
@@ -79,4 +84,47 @@ public void fromIdString_uppercase() {
     public void fromIdString_notExisting() {
         assertFalse(BitcoinNetwork.fromIdString("a.b.c").isPresent());
     }
+
+    @Test
+    public void testLegacyAddressValidity() {
+        LegacyAddress m = LegacyAddress.fromBase58("17kzeh4N8g49GFvdDzSf8PjaPfyoD1MndL", MAINNET);
+        LegacyAddress t = LegacyAddress.fromBase58("n4eA2nbYqErp7H6jebchxAN59DmNpksexv", TESTNET);
+        assertTrue(MAINNET.isValidAddress(m) &&
+                TESTNET.isValidAddress(t) &&
+                SIGNET.isValidAddress(t) &&
+                REGTEST.isValidAddress(t));
```

Comment:
I think single `assertTrue` statements would be easier to read.

Reply:
I'll fix it.

Reply:
Fixed.